File: app/src/main/java/com/example/android/simpleweather/OpenWeatherDataSource.java
package com.example.android.simpleweather;

import com.example.android.simpleweather.dto.CurrentForecast;
import com.example.android.simpleweather.dto.WeatherForecast;
import com.example.android.simpleweather.dto.WeatherForecastHourly;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Query;

interface OpenWeatherDataSource {
    @GET("onecall")
    Call<WeatherForecast> getWeather(@Query("lat") double latitude, @Query("lon") double longitude, @Query("appid") String apiKey);

    @GET("onecall")
    Call<CurrentForecast> getCurrentWeather(@Query("lat") double latitude, @Query("lon") double longitude, @Query("appid") String apiKey);

    @GET("onecall")
    Call<WeatherForecastHourly> getWeatherHourly(@Query("lat") double latitude, @Query("lon") double longitude, @Query("appid") String apiKey);

}

File: app/src/main/java/com/example/android/simpleweather/OpenWeatherDataSource.kt
package com.example.android.simpleweather

import com.example.android.simpleweather.dto.WeatherForecast
import retrofit2.Response
import retrofit2.http.GET
import retrofit2.http.Query

interface OpenWeatherDataSource {
    @GET("onecall")
    suspend fun listRepos(@Query("lat") latitude: Double, @Query("lon") longitude: Double, @Query("appid") apiKey: String): Response<WeatherForecast>
}
